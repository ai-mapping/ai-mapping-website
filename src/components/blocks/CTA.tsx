import React from 'react';
import Section from '@/components/layout/Section';
import Link from 'next/link';

// Main props interface (without form props)
export type CTAProps = {
  heading: string;
  subheading: string;
  buttonText: string;
  buttonLink: string;
};

// Default props
export const defaultCTAProps: CTAProps = {
  heading: 'Ready to Transform Your Development Process?',
  subheading: 'Start using AI-Mapping today and experience the power of AI-driven frontend development.',
  buttonText: 'Get Started',
  buttonLink: '/documentation'
};

const CTA = ({
  heading = defaultCTAProps.heading,
  subheading = defaultCTAProps.subheading,
  buttonText = defaultCTAProps.buttonText,
  buttonLink = defaultCTAProps.buttonLink
}: Partial<CTAProps>) => {
  return (
    <Section background="primary" paddingY="lg" containerWidth="narrow" containerClassName="text-center">
      <h2 className="text-2xl md:text-3xl font-bold text-black mb-4">
        {heading}
      </h2>
      <p className="text-black text-lg mb-8">
        {subheading}
      </p>

      <Link 
        href={buttonLink}
        className="inline-block bg-secondary text-white font-bold px-6 py-3 rounded-lg hover:bg-secondary-light transition-colors duration-300 hover:shadow-lg"
      >
        {buttonText}
      </Link>
    </Section>
  );
};

export default CTA;