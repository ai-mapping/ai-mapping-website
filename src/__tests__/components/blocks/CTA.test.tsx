import React from 'react';
import { render, screen } from '@testing-library/react';
import CTA, { defaultCTAProps } from '@/components/blocks/CTA';

// Create a mock for Next.js Link component since we're using that in our CTA now
jest.mock('next/link', () => {
  return {
    __esModule: true,
    default: ({ children, href, className }) => (
      <a href={href} className={className} data-testid="mock-link">
        {children}
      </a>
    ),
  };
});

describe('CTA Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<CTA />);
    
    // Check if heading and subheading are rendered
    expect(screen.getByText(defaultCTAProps.heading)).toBeInTheDocument();
    expect(screen.getByText(defaultCTAProps.subheading)).toBeInTheDocument();
    
    // Check if the button is rendered
    expect(screen.getByText(defaultCTAProps.buttonText)).toBeInTheDocument();
    expect(screen.getByTestId('mock-link')).toHaveAttribute('href', defaultCTAProps.buttonLink);
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      heading: 'Custom CTA Heading',
      subheading: 'Custom CTA Subheading',
      buttonText: 'Custom Button Text',
      buttonLink: '/custom-link'
    };
    
    const { container } = render(<CTA {...customProps} />);
    
    // Check if custom heading and subheading are rendered
    expect(screen.getByText(customProps.heading)).toBeInTheDocument();
    expect(screen.getByText(customProps.subheading)).toBeInTheDocument();
    
    // Check if the button has custom text and link
    expect(screen.getByText(customProps.buttonText)).toBeInTheDocument();
    expect(screen.getByTestId('mock-link')).toHaveAttribute('href', customProps.buttonLink);
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});