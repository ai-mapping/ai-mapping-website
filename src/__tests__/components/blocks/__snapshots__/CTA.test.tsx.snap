// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-primary py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-3xl text-center"
    >
      <h2
        class="text-2xl md:text-3xl font-bold text-black mb-4"
      >
        Custom CTA Heading
      </h2>
      <p
        class="text-black text-lg mb-8"
      >
        Custom CTA Subheading
      </p>
      <a
        class="inline-block bg-secondary text-white font-bold px-6 py-3 rounded-lg hover:bg-secondary-light transition-colors duration-300 hover:shadow-lg"
        data-testid="mock-link"
        href="/custom-link"
      >
        Custom Button Text
      </a>
    </div>
  </section>
</div>
`;

exports[`CTA Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-primary py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-3xl text-center"
    >
      <h2
        class="text-2xl md:text-3xl font-bold text-black mb-4"
      >
        Ready to Transform Your Development Process?
      </h2>
      <p
        class="text-black text-lg mb-8"
      >
        Start using AI-Mapping today and experience the power of AI-driven frontend development.
      </p>
      <a
        class="inline-block bg-secondary text-white font-bold px-6 py-3 rounded-lg hover:bg-secondary-light transition-colors duration-300 hover:shadow-lg"
        data-testid="mock-link"
        href="/documentation"
      >
        Get Started
      </a>
    </div>
  </section>
</div>
`;
